### PR TITLE
Added pagination and more accurate total to search

### LINKED ISSUE #191: Search returns only 10 results, no pagination
E.g.: https://insights.staging.ubuntu.com/search?q=the - says "We’ve found 10 results for “the”". I suspect that's wrong.

### PR DESCRIPTION
**Pull request by pmahnke**
## Done

- Added pagination and more accurate total to search

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: [a big search](http://0.0.0.0:8023/search?page=7&q=ubuntu)
- See that results look reasonable and the pagination works

## Issue / Card

Fixes #191

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -104,16 +104,19 @@ def category(category_slug):
 @app.route('/search/')
 def search():
     query = flask.request.args.get('q') or ''
-
-    posts = api.search_posts(query) if query else []
+    page = flask.request.args.get('page')
+    print('query: ' + query)
```
Comment (anthonydillon) [inline app.py]: Remove print please.

**Comment by anthonydillon** [comment]
Could you add some validation before querying? For example an empty query results in a 500.

**Comment by anthonydillon** [comment]
Good stuff, just some conflicts.

**Comment by anthonydillon** [comment]
MP: https://code.launchpad.net/~ya-bo-ng/maas/+git/maas/+merge/337210

**Comment by nottrobin** [comment]
This code changes a huge amount in #202. I don't think it's worth getting this PR working. I've created an alternative based off #202:

https://github.com/canonical-websites/insights.ubuntu.com/pull/206

I'll close this now.

**Pull request closed without merging**